code/modules/telescience/radiostation.dm

```diff
@@ -130,59 +130,109 @@
 	icon_state = "mixtable-2"
 	anchored = 1.0
 	density = 1
-	var/state = 0
-	var/state_name = "OFF"
-	var/voice = 0
-	var/last_voice = ""
-
-/obj/submachine/mixing_desk/attack_hand(mob/user as mob)
+	flags = TGUI_INTERACTIVE
+	var/static/list/accents
+	var/list/voices
+	var/selected_voice = 0
+	var/const/max_voices = 9
+	var/say_popup = FALSE
+
+/obj/submachine/mixing_desk/New()
+	. = ..()
+	src.voices = list()
+	if(!src.accents)
+		src.accents = list()
+		for(var/bio_type in concrete_typesof(/datum/bioEffect/speech, FALSE))
+			var/datum/bioEffect/speech/effect = new bio_type()
+			if(!effect.acceptable_in_mutini)
+				continue
+			var/name = effect.id
+			if(length(name) >= 7 && copytext(name, 1, 8) == "accent_")
+				name = copytext(name, 8)
+			name = replacetext(name, "_", " ")
+			accents[name] = effect
+
+/obj/submachine/mixing_desk/ui_state(mob/user)
+	return tgui_default_state
+
+/obj/submachine/mixing_desk/ui_status(mob/user, datum/ui_state/state)
+	return min(
+		state.can_use_topic(src, user),
+		tgui_not_incapacitated_state.can_use_topic(src, user)
+	)
+
+/obj/submachine/mixing_desk/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "MixingDesk", "[src]")
+		ui.open()
+
+/obj/submachine/mixing_desk/ui_data(mob/user)
+	var/list/data = list()
+	data["voices"] = src.voices
+	data["selected_voice"] = src.selected_voice
+	data["say_popup"] = src.say_popup
+	return data
+
+/obj/submachine/mixing_desk/ui_act(action, params, datum/tgui/ui, datum/ui_state/state)
```

Comment:
convention is to not return directly, but set the default return value

Reply:
Waht about if I want to return early? Should I do
```
. = TRUE
return
```
lol

Reply:
no ofc not lol, just for the non-early values, mostly TRUE